### PR TITLE
Disable upload button song and playlist if required fields not filled.

### PR DESCRIPTION
**Pull request by SaurabhGurde**
issue #234
1) Disabled upload button for songs and playlist if required field are not present.
2) added distinguish mark for required field like (*)

### CODE REVIEW

**Comment by AntonioMrtz** [comment]
Hi @SaurabhGurde thanks for your work and time. I will review the PR tomorrow :)

We have a [guide](https://antoniomrtz.github.io/SpotifyElectron/CONTRIBUTING/) on how to contribute. It's recommended to follow the steps for traceability and knowing who's assigned to each issue.

A PR template is also included for consistency between all PRs.Take a look if you have time :)

**Review by AntonioMrtz — CHANGES_REQUESTED** [review]
Code looks good, please check failing pipelines. Build pipeline is supposed to fail on forked repos but fronted tests and format pipelines have to be checked

**Comment by AntonioMrtz** [comment]
Hi @SaurabhGurde , do you think we can the message `Los campos marcados con * son obligatorios.` above the upload button? With this we can be sure the user understand what is * in this context :)

**Comment by SaurabhGurde** [comment]
@AntonioMrtz 
Screenshot from 2024-10-05 22-16-16
Screenshot from 2024-10-05 22-17-19
which one looks better. user will realize fields with (*) mark are mandatory

**Comment by AntonioMrtz** [comment]
> @AntonioMrtz Screenshot from 2024-10-05 22-16-16 Screenshot from 2024-10-05 22-17-19 which one looks better. user will realize fields with (*) mark are mandatory

Hi @SaurabhGurde, both are good actually but I will take second image :). We're working on the internationalization of frontend for including english soon.

**Comment by SaurabhGurde** [comment]
Screenshot from 2024-10-05 22-51-31
Screenshot from 2024-10-05 22-52-23
@AntonioMrtz added label for mandatory field above submit button. plz check latest commit

**Comment by AntonioMrtz** [comment]
> Screenshot from 2024-10-05 22-51-31 Screenshot from 2024-10-05 22-52-23 @AntonioMrtz added label for mandatory field above submit button. plz check latest commit

Code looks good :). 3 tests are failing because of name change of the placeholders. Can we fix them, it should be quick?

**Comment by SaurabhGurde** [comment]
@AntonioMrtz name of the placeholder is same as master branch i just checked recently

**Comment by AntonioMrtz** [comment]
> @AntonioMrtz name of the placeholder is same as master branch i just checked now

Oh cool, tests are passing now

File: Electron/src/components/Sidebar/ModalAddSongPlaylist/Accordion/AddSongPlayListAccordion.tsx [hunk 1]
```diff
@@ -409,12 +417,18 @@ export default function AddSongPlayListAccordion({
...
+                Los campos marcados con * son obligatorios.
+              </p>
 
               <button
                 type="button"
                 onClick={handleSubmitSong}
                 className={`btn btn-lg ${styles.btnSend} d-flex flex-row justify-content-center`}
                 data-testid="sidebar-addsongplaylistaccordion-submit-song"
+                disabled={
+                  !formDataSong.name && !formDataSong.genre && !songFile
```
Comment (AntonioMrtz) [inline Electron/src/components/Sidebar/ModalAddSongPlaylist/Accordion/AddSongPlayListAccordion.tsx:430]: I think this has to be an or statement, we want all the fields to be included

**Review by AntonioMrtz — CHANGES_REQUESTED** [review]

File: Electron/src/components/Sidebar/ModalAddSongPlaylist/Accordion/AddSongPlayListAccordion.tsx (hunk 1 above)
Comment (SaurabhGurde) [reply Electron/src/components/Sidebar/ModalAddSongPlaylist/Accordion/AddSongPlayListAccordion.tsx:430]: > I think this has to be an or statement, we want all the fields to be included. Tests would be good for this case

yes this are all mandatory field, button will be disabled until and unless user fill all the required field. or may be you wanna say something else which i could not understand please elaborate.

Comment (AntonioMrtz) [reply Electron/src/components/Sidebar/ModalAddSongPlaylist/Accordion/AddSongPlayListAccordion.tsx:430]: I checkout your branch to test if it was working correctly. Playlist form worked fine but in form song I found the following:

* I can create a song with ONLY 1 required field filled. We want ALL required fields to be filled in order to enable the button and allow the user to upload a song.
* The disabled condition on that form relies on 'and' statements. The field will be disabled if name field is not filled and song field is not filled and so. We want the logic to be: if name field is not filled or song file is not filled or genre is not selected....

TL;DR -> Disabled logic for song form has to be inversed. Right now you can upload a song with required fields not filled

Comment (SaurabhGurde) [reply Electron/src/components/Sidebar/ModalAddSongPlaylist/Accordion/AddSongPlayListAccordion.tsx:430]: so thats what the logic is is if name, genre, song are filled then only submit button will be enabled. or may be you are trying to say that among name, genre, song only one field should be filled and then button should be enabled right ? so should i convert it from and to or statement

Comment (AntonioMrtz) [reply Electron/src/components/Sidebar/ModalAddSongPlaylist/Accordion/AddSongPlayListAccordion.tsx:430]: All required fields have to filled so the button appears enable

Negating that condition we get :
* Any required field that is not filled should make the button disabled. Your code does exactly the opposite. Launch backend and frontend and test the behaviour of song form, with your code it will let you submit a song with only one required field filled and we don't want that

Comment (SaurabhGurde) [reply Electron/src/components/Sidebar/ModalAddSongPlaylist/Accordion/AddSongPlayListAccordion.tsx]: ok i understood now i have added or condition now button will enabled when all field are present which are required. i checked form from my side as well

**Comment by AntonioMrtz** [comment]
We will be needing tests for this new functionality. Also check our [convention](https://antoniomrtz.github.io/SpotifyElectron/Git-Convention/) for naming branches, PR... PR template is [here](https://github.com/AntonioMrtz/SpotifyElectron/blob/master/.github/pull_request_template.md)

**Comment by AntonioMrtz** [comment]
Hi @SaurabhGurde are you still working on this?

**Comment by SaurabhGurde** [comment]
Yes I am there on this issue. Is there any change needed?

**Comment by AntonioMrtz** [comment]
> Yes I am there on this issue. Is there any change needed?

Yeah, the one i suggested on the coment above:

- Add tests for new functionality
- Check the [convention](https://antoniomrtz.github.io/SpotifyElectron/Git-Convention/)  for contributing: such as branch name, PR description etc

**Comment by SaurabhGurde** [comment]
ok i will change branch name and and raise new PR
